napari/utils/transforms/transform_utils.py

```diff
@@ -466,3 +466,27 @@ def is_matrix_triangular(matrix):
     return is_matrix_upper_triangular(matrix) or is_matrix_lower_triangular(
         matrix
     )
+
+
+def is_diagonal(matrix, tol=1e-8):
+    """Determine whether affine is a diagonal matrix.
+
+    Parameters
+    ----------
+    matrix : 2-D array
+        The matrix to test.
+    tol : float, optional
+        Consider any entries with magnitude < `tol` as 0.
+
+    Returns
+    -------
+    is_diag : bool
+        Boolean indicating whether affine is diagonal.
```

Comment:
```suggestion
        True if matrix is diagonal, False otherwise.
```

Optional: I prefer being explicit about True/False assignments instead of using whether/if, but not required. You probably should change `affine` to `matrix` though.